nav2_costmap_2d/src/costmap_2d_publisher.cpp

```diff
@@ -220,6 +243,9 @@ void Costmap2DPublisher::publishCostmap()
       }
       costmap_update_pub_->publish(std::move(update));
     }
+    if (costmap_raw_update_pub_->get_subscription_count() > 0) {
```

Comment:
Can you break the occupancy grid update population in the lines above this into its own function similar to createRawCostmapUpdateMsg called `createGridUpdateMsg`

Reply:
As agreed at the last working group meeting this work is out of scope of this issue. I have created a separate issue for that: #3956 3956

Reply:
I included that suggestion in the scope of the issue related to this PR and its continuation.
Resolved with bad53b5a122b1ae1af254de3298d233f9972e0a7